packages/@aws-cdk/core/lib/asset-staging.ts

```diff
@@ -15,6 +15,50 @@ import { Stage } from './stage';
 // eslint-disable-next-line
 import { Construct as CoreConstruct } from './construct-compat';
 
+/**
+ * A simple cache class.
+ *
+ * Must be declared at the top of the file because we're going to use it statically in the
+ * AssetStaging class.
+ */
+class Cache<A> {
```

Comment:
Can you move to a separate private file?

Reply:
Well I can, but I didn't because I was afraid people were going to insist I'd write unit tests for it :)

But *sigh* I guess I will then